ado/ipacheckreadreplace.ado

```diff
@@ -148,10 +172,18 @@ qui {
 	else {
 		restore // if there were no selectmultiple variables
 	}
-	* the selectmultiple phase is complete! Now to add in the other corrections:
-	readreplace using `allcorrections', ///
+	******************** add in the other corrections **************************
+	preserve
+	use `allcorrections', clear
+	count 
+	if `r(N)' > 0 {
+		restore
+		readreplace using `allcorrections', ///
 			id(`id') variable(`variable') value(`value') use
-	
+	}
+	else {
+		restore
```

Comment:
I'm going to have to think about this for a minute but it strikes me that in it's current form there seems to be too many `preserve` and `restore` statements. I wonder if a different logical arrangement might lead to less of these pesky calls to create temporary files?